### ISSUE TITLE
documentation form: unable to use domain entity type

### ISSUE THREAD

**Issue by aviv-julienjehannet** [datahub-project/datahub#11525]
**Describe the bug**
When declaring a structured property related to domain, a form cannot present the full list when prompting.

**To Reproduce**
1. Create the structured property `aviv.tableau.dashboard.initialDomain`
2. Create the form `tests.regression.domains`
3. Ingest previous assets
4. Visit the target asset and proceed the form
5. Block with an empty list (see screenshot)

**Code Snippets: Relevant code snippets**
[form.dhub.yaml.txt](https://github.com/user-attachments/files/17247220/form.dhub.yaml.txt)
[property.dhub.yaml.txt](https://github.com/user-attachments/files/17247221/property.dhub.yaml.txt)

**Expected behavior**
We should have the possibility to select any domain from the provided list.
Note: I have already many domains created in my DataHub instance

**Screenshots**
![image](https://github.com/user-attachments/assets/161c1826-76c6-4e18-aee1-e4835e815c01)

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Firefox
 - Version: 130

**Additional context**
- DataHub: 0.14.1
- CLI : 0.14.1

**Logs**
*No specific logs was found.*

**Comment by aviv-julienjehannet** [comment]
My bad. User must type something relevant to select.

**Comment by aviv-julienjehannet** [comment]
The problem occurs when you try to reopen the form. The previous entry recorded in structured property is not properly rendered.
![image](https://github.com/user-attachments/assets/8bce9761-04af-466f-973a-cc3040c58f46)
![image](https://github.com/user-attachments/assets/48750c1e-ad43-4739-99c0-c9238d542311)

Note: sorry for the wrong initial description. The selection is correct; only the rendering in the form is buggy.

**Comment by jjoyce0510** [comment]
Thanks for raising this! We are looking into this actively and will provide an update when we have more details.

**Comment by chriscollins3456** [comment]
hey @aviv-julienjehannet! I'm looking into this issue for you. Can you confirm this isn't an issue for other structured property types on forms or is displaying existing structured property values in the form only broken for this entity type structured property?

I'll be trying to reproduce this. Just as a heads up though, we are bringing lots of great improvements to forms very soon which will certainly solve this problem. It will also include more support for forms (and structured properties) in general to improve the feature for all.

**Comment by aviv-julienjehannet** [comment]
Yes, I confirm it happens only broken when using entity type qualifiers as target and for the related prompt only. Other input fields are ok (and retrieve correct stored values).

Example:

```
  type_qualifier:
    allowed_types:
      - urn:li:entityType:datahub.corpuser
      - urn:li:entityType:datahub.corpGroup
```

When trying to edit from the properties tab, the rendering widget is ok but I re-enter the form, values seems lost although the structured properties do exist

**Comment by aviv-julienjehannet** [comment]
@chriscollins3456 Any news or any ticket or patch to be linked? Did you manage to reproduce the problem?

**Comment by chriscollins3456** [comment]
hey @aviv-julienjehannet ! thanks so much for the info. yes I was able to reproduce this myself which is very helpful. so our plan is to actually merge a lot of improvements to forms and structured properties that would certainly solve this issue for you within the next 2-3 weeks. So i can't guarantee that this fix (and other improvements) will be in the next release, but certainly by the following release.

I appreciate your help here and your patience.

**Comment by aviv-julienjehannet** [comment]
Another pain point related to the current behavior is the impossibility to remap/move domains under parent domain. All move actions are therefore manual :-(
